Added Polish translation

EvidentlyCube opened the pull request:
This PR adds Polish translation to Tiddly Wiki. I made it for personal use but figured there might be worth in making this official - I'd be happy to keep it up to date if it gets merged to the core.

Of the most interesting decisions I made there are two:
 - I decided to keep using the english word `tiddler` to refer to tiddlers because I couldn't think of anything in Polish that would adequately represent how tiny a tiddler can be, and still sounded good and like a proper noun in various contexts.
 - I couldn't quite find good wording for "story river" and "story view" - I am still not sure if they are the same thing. I've opted to go with either `tiddler view` or `main view` equivalents wherever it made sense.

I am publishing it as a draft because I want to take another look and use it for a little bit to see if it sounds good.

pmario commented on languages/pl-PL/icon.tid:
```diff
@@ -0,0 +1,13 @@
+title: $:/languages/en-GB/icon
```
I think this title is wrong. `en-GB` should probably be `pl-PL`

EvidentlyCube replied:
Fixed, I missed it, thank you!

pmario commented:
Why did you "force-push" your changes? A normal push should have worked too.

pmario commented:
>I couldn't quite find good wording for "story river" and "story view" 

"story river" is the area, where the tiddlers are shown. It looks like a river of tiddlers if several of them are open at the same time.

The "story view" is the name of a button in the right sidebar : Tools tab. ... It defines how the "story river" looks like. eg: classic, pop, stacked .. and so on. 

So that's completely different things. I think "tiddler view" and "main view" are _not_ the right terms here.

EvidentlyCube commented:
> Why did you "force-push" your changes? A normal push should have worked too.

Personal preference, but I prefer to amend changes that have no historical value. Since it's in draft and I am still ironing the basics I figured it might as well be amended. I do understand it makes following the changes harder, but I assumed no one will review the code itself at this point, since it's a draft. If you prefer I'll just do normal commits now that it's here?

----

Thanks for the feedback about Story River and Story View!

So the word **Story** is essentially the name for the section of the page, where the open tiddlers are displayed. **Story River** makes sense for use with the default view, where they appear in a big list, though less so for Zoom. And the phrase **Story View** is used for that button because it configures the **View** of the **Story**.

That explains things, I will indeed need to change those phrases. 

The most direct translations would be "Rzeka Historii", "Rzeka Opowieści" or "Rzeka Tiddlerów" but I think those would be word salads for a polish speaker (I might be wrong though, I spend much more time in English interfaces).

I think a sensible option would be to just not translate it and treat it like a proper noun, a part of TiddlyWiki's own vocabulary, like I did with "tiddler". Looking up any information about the wiki or plugins in English would be much easier that way, I see a couple of languages already do that (German, Dutch, Slovak - that one translates it directly to a word meaning "plot" but puts the original english version in parenths), Spanish uses "Chronology", Greek is interesting, goes with something like "Historical Flow", etc...

----

So ultimately my proposal is:
 
 * Use the English version **Story River** and treat it like a proper noun. Cons: An English phrase in Polish translation. Pros: There are hundreds of precedents of  load words in IT and Web in Polish so no one should bat an eye; it IS a TiddlyWiki-specific phrase so it makes sense to keep it as-is; will be easier to understand what a "Story River" is for people who look up info/plugins in English
 * Use **Widok Story River** (ie. Story River view) for the **Story View** button, since that makes most sense given what I said above.

----

Also I have one question - I made some translation notes in a separate file. I think they might be useful for future upkeep, can I just add them as `languages/pl-PL/translation_notes.md`?

pmario commented:
>Also I have one question - I made some translation notes in a separate file. I think they might be useful for future upkeep, can I just add them as languages/pl-PL/translation_notes.md?

If it's short and will help you in the future .. yes. ... But give it a modified date the shows the tiddler at the end of the list ;)

BUT translations in general add to the overall size of the wiki, since they basically duplicate the language strings. ... It isn't that bad, that we need to count bytes, ... but "a plugin here" some "custom CSS there" ... adds up at the end. ...

pmario commented:
>Personal preference, but I prefer to amend changes that have no historical value. Since it's in draft and I am still ironing the basics I figured it might as well be amended. I do understand it makes following the changes harder, but I assumed no one will review the code itself at this point, since it's a draft. If you prefer I'll just do normal commits now that it's here?

I think Jeremy merges a PR with 1 commit only. So your pushes should be easy to follow for others to review. In general I'm only interested in the "diffs". .. 

I don't speak Polish so I only did skim the PR to see it there are some obvious problems. In general it looks good to me

pmario commented:
There isn't much time left till Jeremy will release 5.2.0, that's why I did have a look at the draft. 

see:  TiddlyWiki v5.2.0 release - last call for bug fixes and outstanding PRs #6073

saqimtiaz commented:
> * Use **Widok Story River** (ie. Story River view) for the **Story View** button, since that makes most sense given what I said above.

I have a moderate level of proficiency in Polish and I think this makes sense and is a good approach.

EvidentlyCube commented:
> BUT translations in general add to the overall size of the wiki, since they basically duplicate the language strings. ... It isn't that bad, that we need to count bytes, ... but "a plugin here" some "custom CSS there" ... adds up at the end. ...

Drat, I actually have written a lot. Would it be then fine if I made a thread in Discussions with the translation notes and made a `readme.md` file with the content being essentially:

```
Translation notes are available here: <paste the Discussions link>
```

I just know from experience that if I don't put it somewhere findable it'll get lost D:. And while I **want** to keep this up to date I also know I tend to forget projects and if someone else were to work on this I am pretty sure it would be useful.

> There isn't much time left till Jeremy will release 5.2.0, that's why I did have a look at the draft.

Oh, I didn't even think it could make the cut. It'd be awesome if it did!

pmario commented:
> Oh, I didn't even think it could make the cut. It'd be awesome if it did!

Since I can't read it, I don't know if everything is translated in the right way. .. But if we don't publish it, we will never know ;)

In the OP you wrote:

> This PR adds Polish translation to Tiddly Wiki. I made it for personal use but figured there might be worth in making this official - I'd be happy to keep it up to date if it gets merged to the core.

So if it works for you, without "hurting your eyes", it may work for others too :)

EvidentlyCube commented:
Okay, I think it's ready to be reviewed and merged, I've done all I wanted here.

I've updated the translation, went through all of the internal pages to see if there are any awkward translations, fixed a bunch, added polish icon and readme linking to the discussion thread.

pmario commented:
I did have a look at: https://cdn.evidentlycube.com/prototypes/tiddlywiki-pl/index.html  and it looks good to me. @saqimtiaz ??

EvidentlyCube commented:
Note: After the convo in the discussion thread #6080 I'll be making an update to change `ukryty tiddler` to `tiddler-cień`

Jermolene commented:
Great thanks @EvidentlyCube – if you get a chance to do it in the next few hours we should be able to include the translation in the upcoming v5.2.0 release (currently planned for tomorrow).

EvidentlyCube commented:
@Jermolene  I have just pushed an updated for the shadow tiddlers!

Jermolene commented:
Thanks @EvidentlyCube

pmario commented:
@EvidentlyCube Thank you for your contribution. We do have 32 language settings now. Not all of them are up to date. 

Good work!
